api/tests/Repositories/TestRepositoryTest.php

```diff
@@ -54,6 +57,24 @@ public function testAll()
         $this->assertGreaterThanOrEqual(5, $result->count());
     }
 
+    public function testAllWithLimit()
+    {
+        $entities = factory(App\Models\TestEntity::class, 5)->create();
+
+        $result = $this->repository->all(['*'], null, 2);
+        $this->assertTrue(is_array($result->toArray()));
+        $this->assertGreaterThanOrEqual(2, $result->count());
+    }
+
+    public function testAllWithOffset()
+    {
+        $entities = factory(App\Models\TestEntity::class, 5)->create();
+        $count = $this->repository->count();
+        $result = $this->repository->all(['*'], 2, null);
+        $this->assertTrue(is_array($result->toArray()));
+        $this->assertGreaterThanOrEqual($count-2, $result->count());
```

Comment:
This also needs to be `assertEquals`. If it is greater than, the limit has not been applied.